packages/core/store/slices/__tests__/permissions.spec.tsx

```diff
@@ -425,6 +433,197 @@ describe("permissions slice", () => {
       // Watcher will trigger this
       expect(resolveCalls).toBe(4);
     });
+
+    it("updates if parent changes", async () => {
+      // Given: --------------
+      const parentPermissions = { parent: true };
+      const outsideParentPermissions = { outsideParent: true };
+      const resolvePermissions = jest.fn((_data, params) => {
+        if (params.parent.type === "Parent") {
+          return parentPermissions;
+        }
+        return outsideParentPermissions;
+      });
+
+      const config: Config = {
+        components: {
+          Child: {
+            resolvePermissions,
+            render: () => <div />,
+          },
+          Parent: {
+            fields: {
+              items: { type: "slot" },
+            },
+            render: ({ items: Content }) => <Content />,
+          },
+        },
+      };
+
+      const childComponent = {
+        type: "Child",
+        props: { id: "child-1" },
+      };
+      const parentComponent = {
+        type: "Parent",
+        props: {
+          id: "parent-1",
+          items: [childComponent],
+        },
+      };
+      const initialData = {
+        ...defaultAppState.data,
+        content: [parentComponent],
+      };
+
+      appStore.setState({
+        config,
+        state: walkAppState(
+          {
+            ...defaultAppState,
+            data: initialData,
+          },
+          config
+        ),
+      });
+
+      // When: --------------
+      renderHook(() =>
+        useRegisterPermissionsSlice(appStore, { testGlobal: true })
+      );
+
+      // Wait for the initial resolvePermissions promises to resolve
+      await waitFor(
+        () => Object.keys(appStore.getState().permissions.cache).length > 0
+      );
+
+      await act(async () => {
+        const { dispatch } = appStore.getState();
+
+        // Move the Child component to root
+        dispatch({
+          type: "move",
+          sourceZone: "parent-1:items",
+          sourceIndex: 0,
+          destinationZone: rootDroppableId,
+          destinationIndex: 0,
+        });
+      });
+
+      // Then: --------------
+      expect(resolvePermissions).toHaveBeenCalledTimes(2);
+      expect(resolvePermissions.mock.calls[0][1].parent).toEqual(
+        parentComponent
+      );
+      expect(resolvePermissions.mock.results[0].value).toEqual(
+        parentPermissions
+      );
+
+      expect(resolvePermissions.mock.calls[1][1].parent).toEqual({
+        type: "root",
+        props: { id: "root" },
+      });
+      expect(resolvePermissions.mock.results[1].value).toEqual(
+        outsideParentPermissions
+      );
+    });
```

Comment:
Would it make sense to actually check the updated permissions cache here? Might be easier than checking the mocks